instrumentation/grpc-1.6/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_6/GrpcSingletons.java

```diff
@@ -27,9 +30,14 @@ public final class GrpcSingletons {
         InstrumentationConfig.get()
             .getBoolean("otel.instrumentation.grpc.experimental-span-attributes", false);
 
+    List<String> rpcRequestMetadata =
+        InstrumentationConfig.get()
+            .getList("otel.instrumentation.rpc.capture-metadata.request", emptyList());
```

Comment:
it's not clear to me from the spec if this is a general `rpc` feature or specific to `grpc`?

also, should we split the config into two, `.client.request` and `.server.request`, the way it's split for http?

Reply:
see https://github.com/open-telemetry/opentelemetry-specification/pull/2461 for some history

Reply:
@trask 
This is an rpc feature but this PR only takes care of gRPC

We already discusses the split to client server in one of the resolved thread

Reply:
Oh, I missed that one. Can you rename the property to "otel.instrumentation.**grpc**.capture-metadata.request"?
Since the metadata feature is not generic to all RPC implementations (even if some of them offer similar features), but strictly specific to gRPC.

Reply:
@mateuszrzeszutek @trask 
I changed the config to otel.instrumentation.grpc.capture-metadata.request

Reply:
@trask I read the spec PR you sent me and honestly did not understand the need to separate the headers into client and server, can you please elaborate?

Reply:
it's the configuration which is split into client and server (at least in http), I think so that you can capture a different set of headers on incoming server requests vs the set of headers you capture on outgoing client requests

Reply:
@trask 
So I understand that it requires a change in infrastructure, namely splitting the components to server
and client instrumentation.
This is definetly a good feature, but I think it's a bit out of scope for this PR, perhaps i'll open an issue and take care of it later?

Reply:
hey @Tavh, I'm not sure if we are talking about the same thing, so I sent PR to your branch to clarify what I was imagining if we want to split the config to match the http config: https://github.com/Tavh/opentelemetry-java-instrumentation/pull/1

Reply:
@trask Thanks for the help, I finally get what you meant. Merged the PR :)